neqo-http3/src/connection_client.rs

```diff
@@ -515,6 +516,46 @@ impl Http3Client {
         output
     }
 
+    /// The function establishes a classic HTTP CONNECT tunnel using `target`
+    /// and `headers`. Data can be send into the tunnel via
+    /// [`Http3Client::send_data`] and received from the tunnel via
+    /// [`Http3Client::read_data`]. The tunnel can be closed via
+    /// [`Http3Client::stream_close_send`].
+    ///
+    /// # Errors
+    ///
+    /// If a new stream cannot be created an error will be return.
+    pub fn classic_connect<'x, 't: 'x, T>(
+        &mut self,
+        now: Instant,
+        target: &'t T,
```

Comment:
Is this a case where we want the target to be something else?  An HTTP connect uses a name and port, or an IP and port.  Could we use that here?  We could maybe have the type we choose implement `AsRequestTarget` so that we don't need to tweak `RequestDescription` at all, but a narrower type might avoid misuse of the API.

Reply:
Good point. I added https://github.com/mozilla/neqo/pull/2886/commits/1b6a760e27566e455e92ffad3a223bff8694b9e9 here for now. As a follow-up I created https://github.com/mozilla/neqo/pull/2966.